src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/PseudoCustomAttributesData.cs

```diff
@@ -480,7 +482,7 @@ private static void DecodeMarshalAsCustom(string[] paramNames, object?[] values,
                         name = (string?)values[i];
                         break;
                     case "MarshalTypeRef":
-                        type = (Type?)values[i];
+                        type = Type.GetType((string)values[i]!);
```

Comment:
`Type.GetType` uses runtime type system. What would it take to avoid calling `Type.GetType` here?

(Also, the `UnconditionalSuppressMessage` justification does not look right.)

Reply:
> Type.GetType uses runtime type system. What would it take to avoid calling Type.GetType here?

I guess we can map it to corresponding runtime type, though it says `MarshalAsAttribute.MarshalTypeRef` 
**custom marshaler type** which likely not a runtime type, then if `Type.GetType` only uses runtime type system how I should load the custome type? I thought the custom type (assembly) will be loaded in the context so it can be loaded with `Type.GetType`, no?

Reply:
`Type.GetType` is a static method that always loads the type as the runtime type. It has no way of knowing the context.

There is `Type.GetType` overload where you can supply a custom factory for your assemblies and types. It is something that can be potentially used to implement this. 

Does the Type that this creates leak through the public surface or is it only necessary as an internal implementation detail?

Reply:
> Type.GetType is a static method that always loads the type as the runtime type. It has no way of knowing the context.
> 
> There is Type.GetType overload where you can supply a custom factory for your assemblies and types. It is something that can be potentially used to implement this.

Thanks, good to know!

> Does the Type that this creates leak through the public surface or is it only necessary as an internal implementation detail?

Good point, looks the type is populated when loading the attribute but serialized back as a string during deserialization, so we don't need really need to create the type in between, thank you!